### DISCUSSION TITLE
What is "Resource lifecycle hook"?

### DISCUSSION THREAD

**Discussion by kimminw00** [argoproj/argo-cd#17807] [Q&A]
When deploying helm charts in ArgoCD, an icon ⚓ appears under the k8s object in ArgoCD UI.
If I hover mouse over the icon(⚓), it says “Resource lifecycle hook.”

The anchor icon appears to be displayed when the k8s resources(ex. configmaps and secrets) are created using the [range function](https://helm.sh/docs/chart_template_guide/control_structures/#looping-with-the-range-action) in the helm chart.
For resources having the icon(⚓), auto sync operation did not occur even if the value.yaml files which is related to those are changed.
Is there anyone experiencing the same issue?

**Comment by mateuszbanasiewicz** [comment]
I have the same observation as you. The anchor icon is unclear. 
From ArgoCD UI perspective is not clear the status of k8s resources with used helm or argocd hooks. 
Is there any possibility to improve that in ArgoCD 3?

**Comment by ChristianCiach** [comment]
The anchor icon appears on resources that represent Resource Hooks: https://argo-cd.readthedocs.io/en/stable/user-guide/resource_hooks/

Since you don't appear to be aware of this feature, your Helm chart probably contains [Chart hooks](https://helm.sh/docs/topics/charts_hooks/#the-available-hooks) that are implicitly treated the same as Resource Hooks by Argo CD: https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#helm-hooks

**Reply by mateuszbanasiewicz** [reply]
@ChristianCiach I am aware that I use argocd hooks. My comment is only that the anchor icon is not clear and  doesn't present status of this object. Is it correct applied? or not? 

I consider to migrate hooks feature to wave.
